app/views/maintenance_tasks/tasks/show.html.erb

```diff
@@ -31,14 +31,15 @@
           <%= form.file_field :csv_file %>
         </div>
       <% end %>
-      <% if @task.parameter_names.any? %>
+      <% parameters = @task.parameters %>
+      <% if parameters.any? %>
         <div class="block">
           <%= form.fields_for :task_arguments do |ff| %>
```

Comment:
We can pass an object to `fields_for`, and then we don't need to pass the value for the form helpers. Then we don't need to change `parameter_names` into `parameters`.

Reply:
Definitely an option! The only reason why I haven't implemented it because I was hesitant of making `task_object` publicly accessible 

>Then we don't need to change parameter_names into parameters.

I wonder if with `task_object` being publicly accessible we need the `TaskData#parameter_names` at all? We could just get it from `@task_object.attribute_names` 

^ While I was writing this I realized that it would be better to keep and use `TaskData#parameter_names` method as "parameters" is a maintenance task concept and "attributes" is an implementation detail so it would be better to hide it a little bit

I'll update the form 👍